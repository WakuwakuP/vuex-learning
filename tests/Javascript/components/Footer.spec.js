import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';

import Footer from '@/components/Footer';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('Footer', () => {
  let actions;
  let getters;
  let options;

  beforeEach(() => {
    actions = {
      'auth/logout': jest.fn(),
    };
    getters = {
      'auth/check': () => true,
    };
    const store = new Vuex.Store({
      state: {
        auth: {
          apiStatus: true,
        },
      },
      actions,
      getters,
    });
    options = {
      store,
      localVue,
    };
  });

  // マウントテスト
  it('is a Vue Instance', () => {
    const wrapper = shallowMount(Footer, options);
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  // アクション発火テスト
  it('logout クリック', () => {
    const wrapper = shallowMount(Footer, options);
    wrapper.find('button').trigger('click');
    expect(actions['auth/logout']).toHaveBeenCalled();
  });

  // スナップショットテスト
  it('renders correctly', () => {
    const wrapper = shallowMount(Footer, options);
    expect(wrapper.element).toMatchSnapshot();
  });
});
