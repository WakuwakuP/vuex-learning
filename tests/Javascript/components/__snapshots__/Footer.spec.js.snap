// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<footer
  class="footer"
>
  <button
    class="button button--link"
  >
    Logout
  </button>
</footer>
`;
